api/endpoints/job.py

```diff
@@ -455,29 +455,74 @@ def get(self, job_id):
+    parser.add_argument('job_type', type=str, help="Job type + version (ie. topsapp:v1.0)", required=False)
+    parser.add_argument('tag', type=str, help="User-defined job tag", required=False)
+    parser.add_argument('queue', type=str, help="Submitted job queue", required=False)
+    parser.add_argument('priority', type=int, help="Job priority, 0-9", required=False)
+    parser.add_argument('start_time', type=str, help="Start time of @timestamp field", required=False)
+    parser.add_argument('end_time', type=str, help="Start time of @timestamp field", required=False)
+    parser.add_argument('get_job_details', type=bool, help="Return full details if True. List of job id's if false. Default True.", required=False)
+    parser.add_argument('status', type=str, help="Job status, ie. job-queued, job-started, job-completed, "
+                                                 "job-failed", required=False)
+
 
     @api.doc(security='ApiKeyAuth')
     @login_required
     def get(self, username):
         """
-        This will return run a list of jobs for a specified user
-        :return:
+        Returns a list of jobs for a given user.
+
+        :param username: Username
+        :param get_job_details: Boolean that returns job details if set to True or just job ID's if set to False. Default is True.
+        :param page_size: Page size for pagination
+        :param offset: Offset for pagination
+        :param status: Job status
+        :param end_time: End time
+        :param start_time: Start time
+        :param priority: Job priority
+        :param queue: Queue
+        :param tag: User tag
+        :param job_type: Algorithm type
+        :return: List of jobs for a given user that matches query params provided
         """
-        offset = request.args.get("offset", 0)
-        page_size = request.args.get("page_size", 250)
+        
+        defaults = {
+            "username" : username,
+            "get_job_details" : True, # To preserve existing behavior, set default to True. In the future, we should set this to False.
+            "page_size" : None,
+            "offset" : None,
+            "status" : None,
+            "end_time" : None,
+            "start_time" : None,
+            "priority" : None,
+            "queue" : None,
+            "tag" : None,
+            "job_type" : None
+        }
+        
+        # Get params and set default values
+        params = {key: request.args.get(key, default) for key, default in defaults.items()}
+
+        is_param_true = lambda x: x if isinstance(x, bool) else x.lower() == 'true'
+        get_job_details = is_param_true(params['get_job_details'])
+        
+        # Filter out the non-query params for the Mozart request
+        exclude_list = ["username", "get_job_details"]
+        filtered_query_params = {k: v for k, v in params.items() if k not in exclude_list and v is not None}
+
```

Comment:
Need to add a job status translation somewhere here.

Reply:
Addressed